install.sh

```diff
@@ -289,8 +289,25 @@ GASNET_CC="`$PKG_CONFIG $pkg --variable=GASNET_CC`"
 GASNET_CFLAGS="`$PKG_CONFIG $pkg --variable=GASNET_CFLAGS`"
 GASNET_CPPFLAGS="`$PKG_CONFIG $pkg --variable=GASNET_CPPFLAGS`"
 
-if [ "$GASNET_CC" != "$FPM_CC" ]; then 
-  echo "GASNET_CC=$GASNET_CC" and  "FPM_CC=$FPM_CC don't match"
+# Check whether GASNet was installed using Spack. If yes, bail out.
+# Note: relies on the fact that most Spack installations have "opt/spack" in the directory path
+GASNET_IS_SPACK=$(echo $GASNET_LIBS | awk '{print $1};' | grep -c -v spack)
```

Comment:
The comment should probably also note that the command above assumes the ordering of `$GASNET_LIBS` places the GASNet library directory first, which is currently true but not guaranteed (although also not likely to change).

However I don't understand the `-v` option to `grep` - isn't that backwards?

Reply:
The `-v` option inverts the count such that 0 is true because I thought that's what POSIX `test` / `[` expects (so I can simply put `if [ $cond ]`. But I guess I can remove that since I explicitly test for equality using `-eq`.